final_assignment/micu_matei/installTuxyLab.py

```diff
@@ -0,0 +1,418 @@
+
+    try:
+        with open(path) as data_file:
+            data = json.load(data_file)
+            write_log('o', "Am citit informatiile de configurare din " + path)
+    except ValueError:
+        write_log('e', "Informatiile din fisier nu sunt valide (" + path + ")")
+        data = {'NU_EXISTA': True}
+
+    except IOError:
+        write_log('e', "Nu am gasit fisierul (" + path + ")")
+        data = {'NU_EXISTA': True}
+    return data
+
+
+def parser_cli_arguments():
+    """ parseaza argumentele primite la rularea programului """
+    args = argparse.ArgumentParser()
+    args.add_argument("-c", "--config_file",
+                      help="Path to config file")
+    args = args.parse_args()
+    return args
+
+
+def function_download(name, valori):
+    """ downloadam de la o sursa la o destinatie """
+    it_workerd = True
+    source = valori.get('source', None)
+    destination = valori.get('destination', None)
+    if not source and not destination:
+        it_workerd = False
+        write_log('e', "Nu putem efectua " + name)
+        write_log('d', valori)
+
+    try:
+        response = urllib2.urlopen(source)
+        data = response.read()
+        response.close()
+        try:
+            file_data = open(destination, 'w')
+            file_data.write(data)
+        except IOError as error:
+            write_log('e', 'Nu am putut scrie la locatia ' + destination)
+            write_log('d', error)
+            it_workerd = False
+    except IOError as error:
+        write_log('e', 'Nu am putut descarca de la ' + source)
+        write_log('d', error)
+        it_workerd = False
+
+    write_log('o', "Am descarcat de la <" + source + "> si am ")
+    write_log('o', "salvat in <" + destination + ">")
+    return it_workerd
+
+
+def function_run_script(name, valori):
+    """ setam variabilele de sistem, si rulam un script """
+    attemts = valori['attempts']
+    check_exit_code = valori['check_exit_code']
+
+    if isinstance(check_exit_code, unicode):
+        check_exit_code = bool(check_exit_code)
+    if isinstance(check_exit_code, bool):
+        if check_exit_code:
+            check_exit_code = [0]
+        else:
+            check_exit_code = []
+    env_variables = valori['env_variables']
+    retry_interval = valori['retry_interval']
+    shell = valori["shell"]
+    command = valori["command"]
+
+    cwd = valori['cwd']
+    try:
+        if cwd:
+            os.chdir(cwd)
+        write_log('o', "Am setat cwd :" + cwd)
+    except OSError:
+        write_log('e', "Nu am putut seta cwd :" + cwd)
+        write_log('e', "Probabil nu exista")
+        return False
+
+    for name in env_variables:
+        os.environ[name] = env_variables[name]
+
+    write_log('o', "Am setat variabilele de sistem ")
+    write_log('d', env_variables)
+
+    it_workerd = False
+    while attemts:
+        attemts -= 1
+        if shell:
+            try:
+                ret = os.system(command)
```

Comment:
Am gasit de `subprocess.call` care nu e ceea ce trebuie.FIind in criza de timp am mers cum ce am stiut.